test/e2e/apps/controller_revision.go

```diff
@@ -0,0 +1,253 @@
+		dsList, err := csAppsV1.DaemonSets("").List(context.TODO(), metav1.ListOptions{LabelSelector: dsLabelSelector})
+		framework.ExpectNoError(err, "failed to list Daemon Sets")
+		framework.ExpectEqual(len(dsList.Items), 1, "filtered list wasn't found")
+
+		ds, err := c.AppsV1().DaemonSets(ns).Get(context.TODO(), dsName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+
+		// Listing across all namespaces to verify api endpoint: listAppsV1ControllerRevisionForAllNamespaces
+		ginkgo.By(fmt.Sprintf("Listing all ControllerRevisions with label %q", dsLabelSelector))
+		revs, err := csAppsV1.ControllerRevisions("").List(context.TODO(), metav1.ListOptions{LabelSelector: dsLabelSelector})
+		framework.ExpectNoError(err, "Failed to list ControllerRevision: %v", err)
+		framework.ExpectEqual(len(revs.Items), 1, "Failed to find any controllerRevisions")
+
+		// Locate the current ControllerRevision from the list
+		var initialRevision *appsv1.ControllerRevision
+
+		rev := revs.Items[0]
+		oref := rev.OwnerReferences[0]
+		if oref.Kind == "DaemonSet" && oref.UID == ds.UID {
+			framework.Logf("Located ControllerRevision: %q", rev.Name)
+			initialRevision, err = csAppsV1.ControllerRevisions(ns).Get(context.TODO(), rev.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err, "failed to lookup ControllerRevision: %v", err)
+			framework.ExpectNotEqual(initialRevision, nil, "failed to lookup ControllerRevision: %v", initialRevision)
+		}
+
+		ginkgo.By(fmt.Sprintf("Patching ControllerRevision %q", initialRevision.Name))
+		payload := "{\"metadata\":{\"labels\":{\"" + initialRevision.Name + "\":\"patched\"}}}"
+		patchedControllerRevision, err := csAppsV1.ControllerRevisions(ns).Patch(context.TODO(), initialRevision.Name, types.StrategicMergePatchType, []byte(payload), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch ControllerRevision %s in namespace %s", initialRevision.Name, ns)
+		framework.ExpectEqual(patchedControllerRevision.Labels[initialRevision.Name], "patched", "Did not find 'patched' label for this ControllerRevision. Current labels: %v", patchedControllerRevision.Labels)
+		framework.Logf("%s has been patched", patchedControllerRevision.Name)
+
+		ginkgo.By("Create a new ControllerRevision")
+		newHash, newName := hashAndNameForDaemonSet(ds)
+		newRevision := &appsv1.ControllerRevision{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            newName,
+				Namespace:       ds.Namespace,
+				Labels:          labelsutil.CloneAndAddLabel(ds.Spec.Template.Labels, appsv1.DefaultDaemonSetUniqueLabelKey, newHash),
+				Annotations:     ds.Annotations,
+				OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(ds, appsv1.SchemeGroupVersion.WithKind("DaemonSet"))},
+			},
+			Data:     initialRevision.Data,
+			Revision: initialRevision.Revision + 1,
+		}
+		newControllerRevision, err := csAppsV1.ControllerRevisions(ns).Create(context.TODO(), newRevision, metav1.CreateOptions{})
```

Comment:
I am getting the following error when I am running this localy. You need to change the deployment to create a unique hash of the daemon set to create the controller revision

```
  Jun 10 14:58:35.457: Failed to create ControllerRevision: controllerrevisions.apps "e2e-bd8m8-daemon-set-677bf9f4df" already exists
  Unexpected error:
      <*errors.StatusError | 0xc0039243c0>: {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "controllerrevisions.apps \"e2e-bd8m8-daemon-set-677bf9f4df\" already exists",
              Reason: "AlreadyExists",
              Details: {
                  Name: "e2e-bd8m8-daemon-set-677bf9f4df",
                  Group: "apps",
                  Kind: "controllerrevisions",
                  UID: "",
                  Causes: nil,
                  RetryAfterSeconds: 0,
              },
              Code: 409,
          },
      }
      controllerrevisions.apps "e2e-bd8m8-daemon-set-677bf9f4df" already exists
  occurred
```

Reply:
@atiratree Can we please confirm this error as my local testing doesn't create an error when I test it.

Also, I'm conscious of the time that this test needs to soak before it can be promoted. It would be awesome for all SIG-App endpoints to have conformance tests in this release.

Reply:
I can still reproduce this with `./hack/local-up-cluster.sh`. It fails 100% times. 

Oddly, I am not getting any error with lastest kind.

Can you also please try with `local-up-cluster.sh` and debug the hashes and see what is the difference?